keyboards/sets3n/zx50/keymaps/default/keymap.c

```diff
@@ -0,0 +1,326 @@
+		KC_CAPS, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, KC_L,    KC_SCLN, KC_QUOT, 
+		KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT, 
+		KC_LCTL, KC_LGUI, KC_LALT, KC_NO,   KC_TRNS, TO(2),   KC_SPC,  KC_TRNS, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT),
+// 2:Num Layer    
+    [_NUM] = LAYOUT(
+		KC_TRNS, KC_F1,   KC_F2,   KC_F3,    KC_F4,   KC_F5,   KC_F6,   KC_BSPC,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, 
+		KC_TAB,  KC_Q,    KC_UP,   KC_E,     KC_R,    KC_T,    KC_Y,    KC_LBRC,  KC_P7,   KC_P8,   KC_P9,   KC_PPLS, 
+		KC_CAPS, KC_LEFT, KC_DOWN, KC_RIGHT, RGB_HUD, RGB_SAI, RGB_SAD, KC_QUOT,  KC_P4,   KC_P5,   KC_P6,   KC_PEQL, 
+		KC_LSFT, KC_Z,    KC_X,    KC_C,     KC_V,    KC_B,    KC_N,    KC_ENT,   KC_P1,   KC_P2,   KC_P3,   KC_PCMM, 
+		KC_LCTL, KC_LGUI, KC_LALT, KC_NO,    KC_TRNS, TO(0),   KC_SPC,  KC_TRNS,  KC_P0,   KC_P0,   KC_PDOT, KC_PENT) 
+};
+
+
+// Define RGB Location
+const rgblight_segment_t PROGMEM my_capslock_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+        {0, 1, HSV_MAGENTA}
+);
+
+const rgblight_segment_t PROGMEM my_numlocak_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+        {1, 1, HSV_GREEN}
+);
+
+const rgblight_segment_t PROGMEM my_scroll_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+        {2, 1, HSV_CYAN}
+);
+
+const rgblight_segment_t PROGMEM my_num_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+        {11, 1, HSV_AZURE},
+        {12, 3, HSV_TEAL},
+        {17, 1, HSV_PINK},
+        {23, 3, HSV_GOLDENROD},
+        {26, 1, HSV_TEAL},
+        {28, 3, HSV_PINK},
+        {35, 3, HSV_GOLDENROD},
+        {38, 1, HSV_TEAL},
+        {47, 3, HSV_GOLDENROD},
+        {50, 1, HSV_TEAL},
+        {58, 3, HSV_GOLDENROD},
+        {61, 1, HSV_TEAL}
+);
+// Define the array of layers. Later layers take precedence
+const rgblight_segment_t* const PROGMEM my_rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+        my_capslock_layer,
+        my_numlocak_layer,
+        my_scroll_layer,
+        my_num_layer
+);
+
+void keyboard_post_init_user(void){
+        //Enable the LED layers
+        rgblight_layers = my_rgb_layers;
+}
+
+// Enabling and disabling lighting layers
+layer_state_t layer_state_set_user(layer_state_t state) {
+    // Both layers will light up if both kb layers are active
+    rgblight_set_layer_state(0, layer_state_cmp(state, 0));
+    rgblight_set_layer_state(1, layer_state_cmp(state, 1));
+    rgblight_set_layer_state(2, layer_state_cmp(state, 2));
+    rgblight_set_layer_state(3, layer_state_cmp(state, _NUM));
+
+
+    return state;
+}
+bool led_update_user(led_t led_state) {
+    rgblight_set_layer_state(0, led_state.caps_lock);
+    rgblight_set_layer_state(1, led_state.num_lock);
+    rgblight_set_layer_state(2, led_state.scroll_lock);
+
+    return true;
+}
+
+
+#ifdef WPM_ENABLE
+char wpm_str[10];
+uint16_t wpm_graph_timer = 0;
+#endif
+
+static char current_alpha_oled = '\0';
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
```

Comment:
Just a heads up, this is probably a more efficient way to handle this: 
https://github.com/qmk/qmk_firmware/blob/b5104c3a93942aa531540d5f46bd58cd5b3cdb46/users/drashna/oled/oled_stuff.c#L42-L129

You can use a smaller array, to include only what you need.


Also, ideally, the oled code should be in `zx50.c`, so that there is a default behavior defined.  This is especially important for configurator generated firmware images, as there is no way to add custom code like this.

Reply:
Is it still possible to use？

Reply:
Either version should be fine to use, and end up about the same.